framework/include/utils/MooseEnumItem.h

```diff
@@ -88,7 +95,7 @@ class MooseEnumItem
   std::string _name;
 
   /// The numeric value for item
-  int _id;
+  mutable int _id; // MooseApp::registerExecFlag can change this for ExecFlagTypesg
```

Comment:
The class (Moose::execute_flags) can't update itself, because the s (e.g., EXEC_LINEAR) is a global const that is added to execute_flags. The ID of this global object needs to be incremented as it is inserted into the the execute_flags object.